File: C#/Composite/Department.cs
namespace Composite;

public sealed class Department : IOrganisationUnit
{
    private readonly List<IOrganisationUnit> _units;

    public Department()
        => _units = [];

    public void AddOrganisationUnit(IOrganisationUnit unit)
        => _units.Add(unit);

    public double CalculateTotalSalary()
        => _units.Sum(unit => unit.CalculateTotalSalary());
}

File: typescript/composite/department.ts
import { OrganizationUnit } from "./organization-unit";

export class Department implements OrganizationUnit {
  private organizationUnits: OrganizationUnit[];

  constructor() {
    this.organizationUnits = [];
  }

  addOrganizationUnit(organizationUnit: OrganizationUnit): void {
    this.organizationUnits.push(organizationUnit);
  }

  calculateTotalSalary(): number {
    return this.organizationUnits.reduce(
      (total, unit) => total + unit.calculateTotalSalary(),
      0
    );
  }
}
